src/plugins/intel_gna/src/backend/gna_limitations.hpp

```diff
@@ -284,178 +153,157 @@ class AbstractValidator {
-                           bool exception = true) const override;
-
-    bool ValidateInputPadding(const std::string& name,
-                              const uint32_t pad_h_begin,
-                              const uint32_t pad_h_end,
-                              const uint32_t pad_w_begin,
-                              const uint32_t pad_w_end,
-                              const uint32_t kernel_h,
-                              const uint32_t kernel_w,
-                              const bool throwOnError = true) const override;
-
-    bool ShouldUseOnlyConv2DGnaIface() const override;
-
-    bool ValidateCnn1D(const std::string& name,
-                       const uint32_t inHeight,
-                       const uint32_t inWidth,
-                       const uint32_t inChannels,
-                       const uint32_t kH,
-                       const uint32_t kW,
-                       const uint32_t kN,
-                       const uint32_t strideH,
-                       const uint32_t strideW,
-                       const uint32_t dilationH,
-                       const uint32_t dilationW,
-                       OvGnaType inPrecision,
-                       bool exception = true) const override;
-};
+    constexpr static uint32_t kBufferMaxSize = 65528;
+    constexpr static uint32_t kConvMinFiltersNum = 4;
+    constexpr static uint32_t kConvMaxFiltersNum = 65532;
+    constexpr static uint32_t kConvDilationHeight = 1;
+    constexpr static uint32_t kConvDilationWidth = 1;
+    constexpr static uint32_t kConvFiltersNumDivider = 4;
+    constexpr static uint32_t kConvFilterSizeDivider = 8;
+    constexpr static uint32_t kConvFilterMaxSize = 768;
+    constexpr static uint32_t kConvEachKernelByteAlignment = 16;
+    constexpr static uint32_t kInputByteAlignment = 64;
+    constexpr static uint32_t kNoOfInputsDivisor = 8;
+    constexpr static uint32_t kNoOfInputsLowPrecDivisor = 16;
+    constexpr static uint32_t kAffineMaxBatchSize = 8;
+    constexpr static uint32_t kMaxPoolMaxWindowSize = 6;
+    constexpr static uint32_t kCopyMaxGrouping = 8;
+    constexpr static uint32_t kTransposeMaxSize = 65528;
+    // TODO In the future there should be created class/struct representing all limitations for specific device
+    // versions.
+    constexpr static uint32_t kMaxLayersCountGNA1_0 = 1023;
+    constexpr static uint32_t kMaxLayersCountGNA2_0 = 4096;
+    constexpr static uint32_t kMaxLayersCountGNA3_X = 8192;
+
+    // Currently split layer only supports 2 bytes in int16 and int8 mode.
+    // In fp32 mode this is not necessary but is useful for testing
+    constexpr static uint32_t kBytesPerSplitElement = 2;
+
+    // Currently crop layer only supports 2 bytes in int16 and int8 mode.
+    // In fp32 mode this is not necessary but is useful for testing
+    constexpr static uint32_t kBytesPerCropElement = 2;
+    constexpr static uint32_t kMemoryPageSize = 4096;
+
+    static inline Limitations& GetInstance() {
```

Comment:
My suggestions:
1. What about creating Factory class which would return pointer to the implementation  of Limitations. It would be up to Factory internal implementation if we return pointer to new instance or to the same instance of the class.
2. It would be even better to split Limitations to group of smaller interfaces split by functionality.

Reply:
add 1. We agreed that a singleton implementation is a sufficient solution, consequently allowing only one instance of the plugin to exist at the same time.
add 2. I agree that it would be better, but I would like to limit the number of changes for a single pull request. However, this idea can be considered for future improvements.